ql/src/java/org/apache/hadoop/hive/ql/Driver.java

```diff
@@ -638,46 +654,28 @@ public boolean getResults(List results) throws IOException {
     }
 
     if (isFetchingTable()) {
-      /**
-       * If resultset serialization to thrift object is enabled, and if the destination table is
-       * indeed written using ThriftJDBCBinarySerDe, read one row from the output sequence file,
-       * since it is a blob of row batches.
-       */
-      if (driverContext.getFetchTask().getWork().isUsingThriftJDBCBinarySerDe()) {
-        maxRows = 1;
-      }
-      driverContext.getFetchTask().setMaxRows(maxRows);
-      return driverContext.getFetchTask().fetch(results);
+      return getFetchingTableResults(results);
     }
 
     if (driverContext.getResStream() == null) {
       driverContext.setResStream(context.getStream());
-    }
-    if (driverContext.getResStream() == null) {
-      return false;
+      if (driverContext.getResStream() == null) {
+        return false;
+      }
     }
```

Comment:
This is a little hard to follow.  Perhaps something like:

```
    if (driverContext.getResStream() == null) {
       // If the driver does not have a stream and neither does the context, return
       Stream contextStream = context.getStream();
       if (contextStream == null) {
         return false;
       }
       driverContext.setResStream(contextStream);
      }
    }
```

Reply:
I agree, it's nicer like this, fixed.